src/pages/word-duel.jsx

```diff
@@ -37,44 +38,46 @@ export default function WordDuel() {
         return true;
     }
 
-    /* 
-    TODO: These are very rudamentary rule but they are sufficient for now 
-    There are some corner cases that can occur when there are repeated letters in a guess
-    Case 1: Guess contains multiples of a letter where only 1 is in the word and both "miss"
-        Actual: Both letters are yellow
-        Expected: One letter should be yellow (the first one) and the other should be gray
-    Case 2: Guess contains multiples of a letter where only 1 is in the word where 1 "hits" and 1 "misses"
-        Actual: "Miss" is yellow and "hit" is green
-        Expected: "Miss" is gray and "hit" is green
-    Need to devise a way to make sure each letter is only used once when evaluating a guess
-    */
     function evaluateGuess(guess) {
-        const evaluation = guess.split("").map((letter, idx) => {
-            if (letter === word[idx]) {
-                return EvaluationColor.Green;
-            }
+        const evaluation = new Array(word.length).fill(EvaluationColor.Gray);
+        let wordCopy = word;
 
-            if (word.includes(letter)) {
-                return EvaluationColor.Yellow;
+        for (let i = 0; i < word.length; i++) {
+            if (guess.charAt(i) === word.charAt(i)) {
+                evaluation[i] = EvaluationColor.Green;
+                // make sure to replace with an impossible symbol
+                wordCopy = replaceAt(wordCopy, i, '?');
             }
+        }
 
-            return EvaluationColor.Gray;
-        });
+        for (let i = 0; i < wordCopy.length; i++) {
+            const index = wordCopy.indexOf(guess.charAt(i));
+            if (index > 0) {
+                evaluation[i] = EvaluationColor.Yellow;
+                // make sure to replace with an impossible symbol
+                wordCopy = replaceAt(wordCopy, index, '?');
+            }
+        }
 
         winGame(evaluation);
-
         setEvaluations([...evaluations, evaluation]);
     }
 
     function winGame(evaluation) {
-        if (!evaluation.includes("green")) {
-            return;
+        for (let i = 0; i < evaluation.length; i++) {
```

Comment:
here we go, I figured out what the issue is.  I made a derp that we were looking for a green answer, but this is the results of the five letters.

```javascript
if (evaluation.some(result => result !== "green")) {
    return;
}
```

"If there is anything that isn't green, then it's not done yet"

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
Works perfectly thank you!